src/event/EventPluginHub.js

```diff
@@ -174,7 +180,14 @@ var EventPluginHub = {
    * @param {string} registrationName Name of listener (e.g. `onClick`).
    */
   deleteListener: function(id, registrationName) {
+    var PluginModule =
+      EventPluginRegistry.registrationNameModules[registrationName];
+    if (PluginModule && PluginModule.willDeleteListener) {
+      PluginModule.willDeleteListener(id, registrationName);
+    }
+
     var bankForRegistrationName = listenerBank[registrationName];
+    // TODO: This should never be null -- when is it?
```

Comment:
You mean, if `bankForRegistrationName[id]`? I had that originally but it seemed like `willDeleteListener` should also be called if you switch from a function to null, which means that `EventPluginHub.putListener` probably needs to delegate to `EventPluginHub.deleteListener` if the listener is null… but the `putListener` function in `ReactDOMComponent` also does some enqueueing so now we're having listener deletion happen at a different time depending on whether it's gone completely from props or if it's just falsy…

It seemed easier to make ReactDOMComponent call putListener/deleteListener only when an actual listener is present and have EventPluginHub not worry about falsy values.